private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisher.java

```diff
@@ -82,101 +86,136 @@
 
 public class MavenPublisher {
 
-  private static final Logger LOG = Logger.getLogger(MavenPublisher.class.getName());
-  private static final ExecutorService EXECUTOR = Executors.newFixedThreadPool(1);
+  private static final Logger LOG = LoggerFactory.getLogger(MavenPublisher.class);
   private static final String[] SUPPORTED_SCHEMES = {
-    "file:/", "https://", "gs://", "s3://", "artifactregistry://"
+    "file:/", "http://", "https://", "gs://", "s3://", "artifactregistry://",
+  };
+  private static final String[] SUPPORTED_UPLOAD_SCHEMES = {
+    "file:/", "http://", "https://", "s3://"
   };
-  private static final String[] SUPPORTED_UPLOAD_SCHEMES = {"file:/", "https://", "s3://"};
 
-  public static void main(String[] args)
-      throws IOException, InterruptedException, ExecutionException, TimeoutException {
-    String repo = System.getenv("MAVEN_REPO");
-    boolean publishMavenMetadata = Boolean.parseBoolean(args[3]);
+  public static void main(String[] args) throws Exception {
 
-    if (!isSchemeSupported(repo)) {
+    if (args.length < 4) {
       throw new IllegalArgumentException(
-          "Repository must be accessed via the supported schemes: "
-              + Arrays.toString(SUPPORTED_SCHEMES));
+          "Expected at least 3 arguments: <coordinates> <path to pom> <path to main artifact>"
+              + " <publish maven metadata> [<extra artifacts>]");
     }
 
-    if (!isUploadSchemeSupported(repo)) {
-      throw new IllegalArgumentException(
-          "Repository must be uploaded to via the supported schemes: "
-              + Arrays.toString(SUPPORTED_UPLOAD_SCHEMES));
+    final String repo = System.getenv("MAVEN_REPO");
+    if (Strings.isNullOrEmpty(repo)) {
+      throw new IllegalArgumentException("MAVEN_REPO environment variable must be set");
+    }
+
+    // give friendly warning if true/false not explicitly set as args[3]
+    if (!"true".equalsIgnoreCase(args[3]) && !"false".equalsIgnoreCase(args[3])) {
+      LOG.warn(
```

Comment:
If the value isn't correct, we should bail. The only way this should be called is from our own code anyway, so being strict is fine.